src/main/java/com/walmartlabs/x12/X12Parser.java

```diff
@@ -17,14 +17,37 @@
 
 import com.walmartlabs.x12.exceptions.X12ParserException;
 
+import java.util.Arrays;
+import java.util.List;
+
 public interface X12Parser {
 
     /**
-     * parse the X12 transmission into
-     * a representative Java object
+     * parse the X12 transmission into a representative Java object
      *
      * @return the representative Java object
      * @throws X12ParserException
      */
     <T> T parse(String sourceData);
+
+    /**
+     * parses the source data into a list of segments each line in the source data is a segment
+     */
+    default List<String> splitSourceDataIntoSegments(String sourceData) {
+        return Arrays.asList(sourceData.split("\\r?\\n"));
+    }
+
+    /**
+     * parses the segment into a list of data elements each date element is separated by an asterisk (*)
+     */
+    default List<String> splitSegmentIntoDataElements(String segment) {
+        return Arrays.asList(segment.split("\\*"));
+    }
+
+    /**
+     * extracts the first data element in a segment which is the segment identifier
+     */
+    default String extractSegmentIdentifier(String segment) {
+        return segment.substring(0, segment.indexOf("*"));
```

Comment:
`segment.indexOf("*")` could return `-1` if * is not part of the string. Should we put check?